tests/test_logger.py

```diff
@@ -0,0 +1,86 @@
+"""
+Tests for the ralph.logger module
+"""
+
+from pathlib import Path
+
+from click.testing import CliRunner
+
+import ralph.logger
+from ralph.cli import cli
+from ralph.exceptions import ConfigurationException
+
+
+# pylint: disable=invalid-name, unused-argument
+def test_logger_exists(fs, monkeypatch):
+    """Test the logging system when a correct configuration is provided"""
+
+    mock_default_config = {
+        "version": 1,
+        "propagate": True,
+        "formatters": {
+            "ralph_format": {
+                "format": "%(asctime)-23s %(levelname)-8s %(name)-8s %(message)s"
+            },
+        },
+        "handlers": {
+            "ralph_stderr": {
+                "class": "logging.StreamHandler",
+                "level": "INFO",
+                "stream": "ext://sys.stderr",
+                "formatter": "ralph_format",
+            },
+        },
+        "loggers": {
+            "ralph": {
+                "handlers": ["ralph_stderr"],
+                "level": "INFO",
+            },
+        },
+    }
+
+    Path("dev").mkdir()
+    Path("app/.ralph").mkdir(parents=True)
+
+    monkeypatch.setattr(ralph.logger, "LOG_CONFIG", mock_default_config)
+
+    runner = CliRunner()
+    result = runner.invoke(
+        cli,
+        ["push", "-b", "fs", "test_file"],
+        input="test input",
+    )
+
+    assert result.exit_code == 0
+
+    assert "Pushing archive test_file to the configured fs backend" in result.output
+
+
+# pylint: disable=invalid-name, unused-argument
+def test_logger_no_config(fs, monkeypatch):
+    """Test that an error occurs when no logging configuration exists"""
+
+    mock_default_config = None
+
+    monkeypatch.setattr(ralph.logger, "LOG_CONFIG", mock_default_config)
+
+    runner = CliRunner()
+    result = runner.invoke(cli, ["list", "-b", "fs"])
+
+    assert result.exit_code == 1
+    assert result.exception.__class__ == ConfigurationException
```

Comment:
Use the `pytest.raises` context manager instead: https://docs.pytest.org/en/reorganize-docs/new-docs/user/pytest_raises.html

Reply:
Fixed